src/cdk-experimental/ui-patterns/behaviors/event-manager/event-manager.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * An event that supports modifier keys.
+ */
+export interface EventWithModifiers extends Event {
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
+}
+
+/**
+ * Options that are applicable to all event handlers.
+ */
+export interface EventHandlerOptions {
+  stopPropagation: boolean;
+  preventDefault: boolean;
+}
+
+/**
+ * A config that specifies how to handle a particular event.
+ */
+export interface EventHandlerConfig<T extends Event> extends EventHandlerOptions {
+  handler: (event: T) => Promise<boolean | void>;
+}
+
+/**
+ * Bit flag representation of the possible modifier keys that can be present on an event.
+ */
+export enum ModifierKey {
+  None = 0,
+  Ctrl = 0b1,
+  Shift = 0b10,
+  Alt = 0b100,
+  Meta = 0b1000,
+}
+
+/**
+ * Abstract base class for all event managers.
+ */
+export abstract class EventManager<T extends Event> {
+  private _submanagers: EventManager<T>[] = [];
+
+  protected configs: EventHandlerConfig<T>[] = [];
+  protected beforeFns: ((event: T) => void)[] = [];
```

Comment:
I might leave some comments here around the intent/behavior of these before and after functions.

I also generally push for not using abbreviations in names, e.g. "fns", instead perhaps something like `preHandlers` and `postHandlers`

Reply:
These were removed